src/Pos/Block/Network/Listeners.hs

```diff
@@ -78,24 +79,23 @@ stubListenerConv' = unproxy $ \(_ :: Proxy ssc) ->
 handleGetHeaders
     :: forall ssc m.
        (WorkMode ssc m)
-    => (ListenerSpec m, OutSpecs)
-handleGetHeaders = listenerConv $ \_ peerId conv -> do
-    logDebug $ "handleGetHeaders: request from " <> show peerId
-    handleHeadersCommunication conv
+    => m (ListenerSpec m, OutSpecs)
+handleGetHeaders = reifyMsgLimit (Proxy @MsgGetHeaders) $ \limitProxy ->
```

Comment:
It's possible bug actually. We reify to concrete message size, i.e. a number. But if it's to be taken from monadic context, it's dynamic, i.e. determined differently at each particular point of runtime